src/kbmod/analysis_utils.py

```diff
@@ -372,325 +358,213 @@ def load_results(
-                    it should have at least 'psi_curves', 'phi_curves', and
-                    'results'. These are populated in Interface.load_results().
-                filter_params : dictionary
-                    A dictionary of additional filtering parameters. Must
-                    contain sigmaG_filter_type.
-            OUTPUT-
-                keep_idx_results : list
-                    list of tuples containing the index of a results, the
-                    indices of the passing values in the lightcurve, and the
-                    new likelihood for the lightcurve.
+
+        Arguments:
+            result_list (`ResultList`): The values from trajectories. This data gets modified directly
+                by the filtering.
+            filter_params (dict): A dictionary of additional filtering parameters. Must
+                contain sigmaG_filter_type.
         """
         print("Applying Clipped-sigmaG Filtering")
         self.lc_filter_type = filter_params["sigmaG_filter_type"]
         start_time = time.time()
-        # Make copies of the values in 'old_results' and create a new dict
-        psi_curves = np.copy(old_results["psi_curves"])
-        psi_curves[np.isnan(psi_curves)] = 0.0
-        phi_curves = np.copy(old_results["phi_curves"])
-        phi_curves[np.isnan(phi_curves)] = 1e9
 
+        # Compute the coefficients for the filtering.
         if self.coeff is None:
             if self.sigmaG_lims is not None:
                 self.percentiles = self.sigmaG_lims
             else:
                 self.percentiles = [25, 75]
             self.coeff = self._find_sigmaG_coeff(self.percentiles)
 
-        num_curves = len(psi_curves)
-        index_list = [j for j in range(num_curves)]
-        zipped_curves = zip(psi_curves, phi_curves, index_list)
-        keep_idx_results = []
         if self.num_cores > 1:
+            zipped_curves = result_list.zipped_curve_list()
+
+            keep_idx_results = []
             print("Starting pooling...")
             pool = mp.Pool(processes=self.num_cores)
             keep_idx_results = pool.starmap_async(self._clipped_sigmaG, zipped_curves)
             pool.close()
             pool.join()
             keep_idx_results = keep_idx_results.get()
+
+            for i, res in enumerate(keep_idx_results):
+                result_list.results[i].filter_indices(res[1])
         else:
-            for z in zipped_curves:
-                keep_idx_results.append(self._clipped_sigmaG(z[0], z[1], z[2]))
+            for i, row in enumerate(result_list.results):
+                single_res = self._clipped_sigmaG(row.psi_curve, row.phi_curve, i)
+                row.filter_indices(single_res[1])
 
         end_time = time.time()
         time_elapsed = end_time - start_time
         print("{:.2f}s elapsed".format(time_elapsed))
         print("Completed filtering.", flush=True)
         print("---------------------------------------")
```

Comment:
I had to go look up what `zip_curves_list` does, which means it could probably have a better name. I guess it didn't occur to me in the other PR.

If it's not too big of a bother, it only appears, perhaps `zip_phi_psi_idx`? It's clearer and shorter. Bonus points if you return a generator instead of a list (just use `()` instead of `[]` in return) to stay true to name; I'm not sure `starmap_async` would notice the difference and there could be a small perf gain.

This entire phrase also repeats itself verbatim in `apply_clipped_average` so I'd suggest replacing it with a private generic method like:

```python
def __parallel_apply_statistic(self, result_list, statistic):
    start_time = time.time()

    if self.num_cores > 1:
        zipped_curves = result_list.zip_phi_psi_idx()

        keep_idx_results = []
        print("Starting pooling...")
        pool = mp.Pool(processes=self.num_cores)
        keep_idx_results = pool.starmap_async(statistic, zipped_curves)
        pool.close()
        pool.join()
        keep_idx_results = keep_idx_results.get()

        for i, res in enumerate(keep_idx_results):
            result_list.results[i].filter_indices(res[1])
    else:
        for i, row in enumerate(result_list.results):
            single_res = statistic(row.psi_curve, row.phi_curve, i)
            row.filter_indices(single_res[1])

    # TODO: get actual logging 
    end_time = time.time()
    time_elapsed = end_time - start_time
    print("{:.2f}s elapsed".format(time_elapsed))
    print("Completed filtering.", flush=True)
    print("---------------------------------------")

    return result_list
```
and then these parts of `apply_clipped_sigmaG` and `apply_clipped_average` are just:
```python
# TODO: replace with logging
print("Applying Clipped-sigmaG Filtering")
self.lc_filter_type = filter_params["sigmaG_filter_type"]
results_list = self.__parallel_apply_statistic(results_list, self._clipped_sigmaG)

```
and
```python
# TODO: replace with logging
print("Applying Clipped-Average Filtering")
self.lc_filter_type = filter_params["sigmaG_filter_type"]
results_list = self.__parallel_apply_statistic(results_list, self._clipped_average)
```

Reply:
I changed the name. I'm holding off on combining the two functions because we were discussion removing everything except sigmaG filtering anyway.